arangod/Aql/Functions.cpp

```diff
@@ -5342,6 +5342,117 @@ AqlValue Functions::GeoPolygon(arangodb::aql::Query* query,
   return AqlValue(b);
 }
 
+// TODO: merge MULTIPOLYGON with POLYGON and reuse functions
+/// @brief function GEO_MULTIPOLYGON
+AqlValue Functions::GeoMultiPolygon(arangodb::aql::Query* query,
+                               transaction::Methods* trx,
+                               VPackFunctionParameters const& parameters) {
+  size_t const n = parameters.size();
+
+  if (n < 1) {
+    // no parameters
+    return AqlValue(AqlValueHintNull());
+  }
+
+  AqlValue const& geoArray = extractFunctionParameterValue(parameters, 0);
+
+  if (!geoArray.isArray()) {
+    ::registerWarning(query, "GEO_MULTIPOLYGON",
+                      TRI_ERROR_QUERY_ARRAY_EXPECTED);
+    return AqlValue(arangodb::velocypack::Slice::nullSlice());
+  }
+
+  VPackBuilder b;
+  b.openObject();
+  b.add("type", VPackValue("MultiPolygon"));
+  b.add("coordinates", VPackValue(VPackValueType::Array));
+
+  AqlValueMaterializer materializer(trx);
+  VPackSlice s = materializer.slice(geoArray, false);
+
+  /*
+  return GEO_MULTIPOLYGON([
+    [
+       [[40, 40], [20, 45], [45, 30], [40, 40]]
+    ],
+    [
+        [[20, 35], [10, 30], [10, 10], [30, 5], [45, 20], [20, 35]],
+        [[30, 20], [20, 15], [20, 25], [30, 20]]
+    ]
+  ])
+  */
+
+  if (s.isArray() && s.length() < 2) {
```

Comment:
I think this check is wrong:
It should trigger if `s` is not an Array, or length is smaller then 2:
`!s.isArray() || s.length() < 2`